occurrence-ws/src/main/java/org/gbif/occurrence/ws/config/WebMvcConfig.java

```diff
@@ -102,30 +111,45 @@ public Object postProcessAfterInitialization(@NotNull Object bean, String beanNa
           argumentResolvers.add(0, paramNameProcessor());
           adapter.setArgumentResolvers(argumentResolvers);
         }
+        if (bean instanceof AbstractJackson2HttpMessageConverter) {
+          AbstractJackson2HttpMessageConverter converter = (AbstractJackson2HttpMessageConverter) bean;
+          ObjectMapper objectMapper = converter.getObjectMapper();
+          objectMapper.registerModule(new JavaTimeModule());
+
+          objectMapper.registerModule(new SimpleModule()
+            .addKeyDeserializer(OccurrenceSearchParameter.class, new OccurrenceSearchParameter.OccurrenceSearchParameterKeyDeserializer())
+            .addDeserializer(OccurrenceSearchParameter.class, new OccurrenceSearchParameter.OccurrenceSearchParameterDeserializer())
+          );
+          objectMapper.addMixIn(SearchParameter.class, EsQueryVisitorFactory.OccurrenceSearchParameterMixin.class);
```

Comment:
I think we can use the one that exists in occurrence-common

Reply:
and delete the one from EsQueryVisitorFactory (in dev and master it's deleted already)

Reply:
done